### PR TITLE
client: add definition for *48_r on musl

### PR DESCRIPTION
**Pull request by Cogitri**
Unfortunately musl doesn't provide these

### CODE REVIEW

**Comment by Cogitri** [comment]
Alternatively instead of using `mrand48_r` and `seed48_r`, `jrand48` could be used which is POSIX and also threadsafe.

**Comment by dvdhrm** [comment]
Yeah, if musl provides jrand48, lets switch to it.

I am unsure why `jrand48_r()` even exists. Looking at the glibc implementation, `jrand48()` is already re-entrant-safe. Probably some posix'ism. If musl has `jrand48_r()`, I would use it for consistency reason. Otherwise, I'd be fine with `jrand48()` as well.

**Comment by Cogitri** [comment]
Alright, I've pushed a new version using jrand4.

**Comment by thom311** [comment]
Odd, from the reading of `man jrand48`, it's not clear to me that the function is really thread safe. Anyway, it probably wouldn't make sense otherwise in any sensible implementation.

**Comment by Cogitri** [comment]
According to POSIX[0] they have to be thread safe:

> By using different arguments, erand48(), nrand48() and jrand48() allow separate modules of a large program to generate several independent streams of pseudo-random numbers, that is the sequence of numbers in each stream will not depend upon how many times the routines are called to generate numbers for the other streams.

However, at least in the glibc implementation `{e,n,j}rand48()` seem to access the same (global) polynomial that is set by e.g. `lcong48`.[1] As such we should be careful not to mix the usage of thread unsafe and thread safe (pseudo) random number generators. As such this shouldn't pose a problem as long as only the thread safe variants are used.

0: https://pubs.opengroup.org/onlinepubs/007908799/xsh/drand48.html
1: https://sourceware.org/bugzilla/show_bug.cgi?id=10320

File: src/n-dhcp4-c-probe.c [hunk 1]
```diff
@@ -375,10 +373,8 @@ static void n_dhcp4_client_probe_config_initialize_random_seed(NDhcp4ClientProbe
  */
 uint32_t n_dhcp4_client_probe_config_get_random(NDhcp4ClientProbeConfig *config) {
         long int result;
-        int r;
 
-        r = mrand48_r(&config->entropy, &result);
-        c_assert(!r);
+        result = jrand48(config->entropy);
 
         return result;
```
Comment (dvdhrm) [inline src/n-dhcp4-c-probe.c:379]: Why not just:

```c
        return jrand48(config->entropy);
```

**Comment by Cogitri** [comment]
Oh right, how did I not see that :D

Thanks for the review, will fix in a bit :)

File: src/n-dhcp4-c-probe.c (hunk 1 above)
Comment (Cogitri) [reply src/n-dhcp4-c-probe.c]: Oh, how did I not see that :D 

Thanks, fixed.

**Comment by thom311** [comment]
merged to master. thank you!!

**Pull request closed without merging**